indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -961,7 +961,12 @@ public Map<String, List<Interval>> getLockedIntervals(List<LockFilterPolicy> loc
 
             final int priority = lockFilter.getPriority();
             final boolean ignoreAppendLocks =
-                TaskLockType.REPLACE.name().equals(lockFilter.getContext().get(Tasks.TASK_LOCK_TYPE));
+                Boolean.TRUE.equals(lockFilter.getContext().getOrDefault(
+                    Tasks.USE_CONCURRENT_LOCKS,
+                    Tasks.DEFAULT_USE_CONCURRENT_LOCKS))
+                || TaskLockType.REPLACE.name().equals(lockFilter.getContext().getOrDefault(
+                    Tasks.TASK_LOCK_TYPE,
+                    Tasks.DEFAULT_TASK_LOCK_TYPE));
```

Comment:
Slightly reformatted for better readability since this condition has become a little difficult now.

```suggestion
                Boolean.TRUE.equals(
                    lockFilter.getContext().getOrDefault(
                        Tasks.USE_CONCURRENT_LOCKS,
                        Tasks.DEFAULT_USE_CONCURRENT_LOCKS
                    )
                )
                || TaskLockType.REPLACE.name().equals(
                    lockFilter.getContext().getOrDefault(
                        Tasks.TASK_LOCK_TYPE,
                        Tasks.DEFAULT_TASK_LOCK_TYPE
                    )
                );
```

You may also consider breaking up the condition.
```
final boolean isUsingConcurrentLocks = condition1;
final boolean isReplaceLock = condition2;
final boolean ignoreAppendLocks = isUsingConcurrentLocks || isReplaceLock;
```

Reply:
Thanks, I've broken up the condition.
I've also verified that the USE_CONCURRENT_LOCKS flag is being used wherever TASK_LOCK_TYPE is, and also that it supersedes the latter.